### PR TITLE
feat(sl-5): resolve Exercice 2 (frere/oncle LTN) into guided example + variation

### PR DESCRIPTION
**Pull request by jsboige**
## Summary

See #2161 (SymbolicLearning exercises → Exemples guidés). Resolves **SL-5 Exercice 2** into a worked example following the proven SL-7/8/10 recipe (solve stub → "Exemple guide", keep `# Indice`/`# Etape`, add a C.1-clean variation exercise).

## Changes (1 file, +170/-12)

- **Ex2** (cell `b7c8387b`): stub → **Exemple guide 2**.
  - Trains `frere(X,Y)` on facts (Marie/Sophie sisters) — reuses the `NeuralPredicate` + cross-entropy idiom of the `parent` cell.
  - Derives `oncle(X,Z)` via `frere(X,Y) AND parent(Y,Z) => oncle(X,Z)` — reuses the rule-guided idiom of the `grandparent` cell + the already-trained `parent_pred`.
- **Variation** (new, after Ex2): "Amitié et transitivité discutable" — train `ami`, derive `ami_ami` by `ami(X,Y) AND ami(Y,Z) => ami_ami(X,Z)`; discusses why rule-learning is soundness-blind. C.1-clean stub.
- **Ex3/Ex4/Ex5**: `execution_count` bumped +1 (insertion hygiene; source + outputs unchanged).

## Proof of execution (kernel python3)

```
frere(Marie, Sophie) = 0.9677   [positif ~1]
frere(Sophie, Marie) = 0.9795   [positif ~1]
frere(*) negatives   = 0.0002..0.0140   [negatifs ~0]
oncle(Marie, Luc)    = 0.8340   [body: frere 0.968 AND parent 0.914]
oncle(Sophie, Pierre)= 0.8317   [body: frere 0.980 AND parent 0.897]
```

## Validation

- `nbformat.validate` OK. **C.1 = 0** (no `raise NotImplementedError`/`assert False`/`1/0`). **H.3 = 0** (no code cell with `execution_count=None` + no outputs).
- `execution_count` monotonic [1..13].
- **Scope**: only Ex2 (source+outputs), the 2 new variation cells, and Ex3/4/5 exec_count bumps. Section-7 LTNtorch cell (`4b4e4ecb`) and all infra cells are **byte-identical to origin/main**.

## Env note (rule F)

SL-5 section 7 uses **LTNtorch** (torch-based `ltn`, has `ltn.Connective`; the PyPI `ltn` is the wrong TF-based package). Installed LTNtorch from `github.com/logictensornetworks/LTNtorch`. Section 7 then hits a CUDA device-mismatch on this box (`cuda.is_available()=True` but mixed device tensors; `CUDA_VISIBLE_DEVICES=''` does not disable CUDA here) — a **pre-existing GPU/device cell unrelated to Ex2** (pure-numpy). Per the rule-F GPU exception, Ex2 was re-executed via its pure-numpy dependency chain (cells 2,5,8,11,14) and section 7 left with its valid origin/main outputs. Flagged to ai-01 for a GPU-machine full-papermill pass (cf. the swi-prolog gap on SL-4/SL-9).

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

**Comment by clusterManager-Myia** [comment]
**[NanoClaw]** — LGTM

**SL-5 Ex.2 (frere/oncle LTN) — stub converti en exemple guide + variation exercise.**

1 file (+170/-12), un seul notebook.

**Exemple guide :** Entrainement de `frere(X,Y)` par faits positifs/negatifs (entropie croisee, lr=0.5, 150 epochs), puis derivation de `oncle(X,Z)` via la regle `frere(X,Y) AND parent(Y,Z) => oncle(X,Z)` (fuzzy implication, 30 epochs). Output authentique : loss decroissante (0.58 -> 0.12 pour frere, 0.37 -> 0.12 pour oncle), predictions coherentes (positifs ~0.97, negatifs ~0.01).

**Variation exercise :** `ami(X,Y)` / `ami_ami(X,Z)` transitivite discutable. Bonne question pedagogique : pourquoi la transitivite de l"amitie est semantiquement discutable ? (rule-gated learning vs truth). Cellule stub avec `execution_count: 10` et prints reels.

**Checks:**
- Output cells authentiques (loss curves, prediction values — pas N/A)
- `execution_count` present (reels)
- Pas de secrets
- Reuse `parent_pred` deja entraine (bonne continuite)
- Pattern identique a SL-7/8/10 recipe
- Renumbering execution_count 10->11, 11->12, 12->13 (coherent)

Clean.
